src/bridges/angle/AngleBridge.sol

```diff
@@ -0,0 +1,125 @@
+        poolManagers[0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48] = 0xe9f183FC656656f1F17af1F2b0dF79b8fF9ad8eD;
+        poolManagers[0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2] = 0x3f66867b4b6eCeBA0dBb6776be15619F73BC30A2;
+        poolManagers[0x853d955aCEf822Db058eb8505911ED77F175b99e] = 0x6b4eE7352406707003bC6f6b96595FD35925af48;
+
+        // _preApprove(0xc9daabC677F3d1301006e723bD21C60be57a5915);
+        // _preApprove(0xe9f183FC656656f1F17af1F2b0dF79b8fF9ad8eD);
+        // _preApprove(0x3f66867b4b6eCeBA0dBb6776be15619F73BC30A2);
+        // _preApprove(0x6b4eE7352406707003bC6f6b96595FD35925af48);
+    }
+
+    /**
+     * @notice A function which returns an _totalInputValue amount of _inputAssetA
+     * @param _inputAssetA - ERC20/ETH (deposit), or sanToken (withdraw)
+     * @param _outputAssetA - sanToken (deposit), or ERC20/ETH (withdraw)
+     * @param _totalInputValue - amount of ERC20/ETH to deposit, or the amount of sanToken to withdraw
+     * @param _auxData - 0 (deposit), 1 (withdraw)
+     * @return outputValueA - the amount of output asset to return
+     */
+    function convert(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        AztecTypes.AztecAsset memory,
+        uint256 _totalInputValue,
+        uint256,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidInputA();
+        if (_outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidOutputA();
+
+        IERC20(_inputAssetA.erc20Address).approve(address(STABLE_MASTER), _totalInputValue);
+
+        if (_auxData == 0) {
+            address poolManager = poolManagers[_inputAssetA.erc20Address];
+            if (poolManager == address(0)) revert ErrorLib.InvalidInputA();
+
+            (, address sanToken, , , , , , , ) = STABLE_MASTER.collateralMap(poolManager);
+            if (sanToken != _outputAssetA.erc20Address) revert ErrorLib.InvalidOutputA();
+
+            STABLE_MASTER.deposit(_totalInputValue, address(this), poolManager);
+        } else if (_auxData == 1) {
+            address poolManager = poolManagers[_outputAssetA.erc20Address];
+            if (poolManager == address(0)) revert ErrorLib.InvalidOutputA();
+
+            (, address sanToken, , , , , , , ) = STABLE_MASTER.collateralMap(poolManager);
+            if (sanToken != _inputAssetA.erc20Address) revert ErrorLib.InvalidInputA();
+
+            STABLE_MASTER.withdraw(_totalInputValue, address(this), address(this), poolManager);
+        }
+
+        outputValueA = IERC20(_outputAssetA.erc20Address).balanceOf(address(this));
```

Comment:
We can do a gas optimization here. If you would subtract dust amount (typically 1) from the balance then the storage slot would never reset to 0 which would make it more efficient to use. This is caused by [EIP-1087](EIP-1087). We use this optimization for example in the [Liquity's StakingBridge](https://github.com/AztecProtocol/aztec-connect-bridges/blob/6388b08be07ebd4d86a2de8e6bf2ca3c2cc8ac5b/src/bridges/liquity/StakingBridge.sol#L174).

Reply:
thanks for the tip!